test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/ArrayTest.scala

```diff
@@ -7,85 +7,91 @@
-        propString += item
-      }
+  @Test def should_provide_implicit_conversion_from_js_Array_to_ArrayOps_Int(): Unit = {
+    var propCount = 0
+    var propString = ""
 
-      expect(propCount).toEqual(4)
-      expect(propString).toEqual("7357")
+    for (item <- Array(7, 3, 5, 7)) {
+      assertTrue(item.isInstanceOf[Int])
+      propCount += 1
+      propString += item
     }
 
-    it("should provide implicit conversion from js.Array to ArrayOps - Char") {
-      var propCount = 0
-      var propString = ""
+    assertEquals(4, propCount)
+    assertEquals("7357", propString)
+  }
 
-      for (item <- js.Array('S', 'c', 'a', 'l', 'a')) {
-        expect(item.isInstanceOf[Char]).toBeTruthy
-        propCount += 1
-        propString += item
-      }
+  @Test def should_provide_implicit_conversion_from_js_Array_to_ArrayOps_Char(): Unit = {
+    var propCount = 0
+    var propString = ""
 
-      expect(propCount).toEqual(5)
-      expect(propString).toEqual("Scala")
+    for (item <- Array('S', 'c', 'a', 'l', 'a')) {
+      assertTrue(item.isInstanceOf[Char])
+      propCount += 1
+      propString += item
     }
 
-    it("should provide implicit conversion from js.Array to ArrayOps - value class") {
-      var propCount = 0
-      var propString = ""
+    assertEquals(5, propCount)
+    assertEquals("Scala", propString)
+  }
 
-      for (item <- js.Array(new VC(5), new VC(-4))) {
-        expect(item.isInstanceOf[VC]).toBeTruthy
-        propCount += 1
-        propString += item
-      }
+  @Test def should_provide_implicit_conversion_from_js_Array_to_ArrayOps_value_class(): Unit = {
+    var propCount = 0
+    var propString = ""
 
-      expect(propCount).toEqual(2)
-      expect(propString).toEqual("VC(5)VC(-4)")
+    for (item <- Array(new VC(5), new VC(-4))) {
+      assertTrue(item.isInstanceOf[VC])
+      propCount += 1
+      propString += item
     }
 
+    assertEquals(2, propCount)
+    assertEquals("VC(5)VC(-4)", propString)
   }
 
-  describe("scala.scalajs.js.JSConverters.JSRichGenTraversableOnce") {
+  // scala.scalajs.js.JSConverters.JSRichGenTraversableOnce
 
+  @Test def should_provide_toJSArray(): Unit = {
     import js.JSConverters._
-
-    it("should provide toJSArray") {
-      expect(List("foo", "bar").toJSArray).toEqual(js.Array("foo", "bar"))
-      expect(Iterator(1, 2, 3).toJSArray).toEqual(js.Array(1, 2, 3))
-      expect(Array(0.3, 7.3, 8.9).toJSArray).toEqual(js.Array(0.3, 7.3, 8.9))
-      expect(None.toJSArray).toEqual(js.Array())
-      // The following fails on 2.10.x
-      //expect(Some("Hello World").toJSArray).toEqual(js.Array("Hello World"))
-    }
-
+    assertArrayEquals(Array("foo", "bar"), List("foo", "bar").toJSArray.toArray)
+    assertArrayEquals(Array(1, 2, 3), Iterator(1, 2, 3).toJSArray.toArray)
+    assertArrayEquals(Array(0.3, 7.3, 8.9), Array(0.3, 7.3, 8.9).toJSArray.toArray)
+    assertArrayEquals(Array.empty[AnyRef], None.toJSArray.asInstanceOf[js.Array[Any]].toArray)
```

Comment:
I think we really should have an `assertJSArrayEquals` defined somewhere, and use that instead. There are most likely a *lot* of cases in this PR where we expect the right `js.Array`.